saleor/csv/utils/export.py

```diff
@@ -0,0 +1,156 @@
+from tempfile import NamedTemporaryFile
+from typing import IO, TYPE_CHECKING, Dict, List, Set, Union
+
+import petl as etl
+from django.utils import timezone
+
+from ...product.models import Product
+from .. import FileTypes
+from ..emails import send_email_with_link_to_download_file, send_export_failed_info
+from .products_data import get_export_fields_and_headers_info, get_products_data
+
+if TYPE_CHECKING:
+    # flake8: noqa
+    from django.db.models import QuerySet
+    from ..models import ExportFile
+
+
+BATCH_SIZE = 10000
+
+
+def export_products(
+    export_file: "ExportFile",
+    scope: Dict[str, Union[str, dict]],
+    export_info: Dict[str, list],
+    file_type: str,
+    delimiter: str = ";",
+):
+    file_name = get_filename("product", file_type)
+    queryset = get_product_queryset(scope)
+
+    export_fields, file_headers, data_headers = get_export_fields_and_headers_info(
+        export_info
+    )
+
+    create_file_with_headers(file_headers, delimiter, export_file, file_name, file_type)
+
+    export_products_in_batches(
+        queryset,
+        export_info,
+        set(export_fields),
+        data_headers,
+        delimiter,
+        export_file,
+        file_type,
+    )
+
+
+def get_filename(model_name: str, file_type: str) -> str:
+    return "{}_data_{}.{}".format(
+        model_name, timezone.now().strftime("%d_%m_%Y"), file_type
```

Comment:
Are you sure that `"%d_%m_%Y"` will be enough? What about the case when I want to export two lists of products on the same day? This method will return the same path to file and will overwrite the previous one. Am I right? Is it expected behavior?

Reply:
No, files won't be overridden. Django always creates unique names like `product_data_01_06_2020_Y9pH4kO.csv`